src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,53 @@
+.calendar-wrapper {
+  height: 100vh;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
+
+.calendar {
+  width: $calendar-width;
+  display: flex;
+  flex-direction: row;
+  flex-wrap: wrap;
+  gap: $cell-gap;
+  padding: $calendar-padding;
+
+  @each $day, $number in $days-of-week {
+    &--start-day--#{$day} :first-child {
+      margin-left: ($cell-size + $cell-gap) * $number;
+    }
+  }
+
+  @for $index from 28 through 31 {
+    &--month-length--#{$index} :nth-child(n + #{$index + 1}) {
+      display: none;
+    }
+  }
+
+  &__cell {
+    display: flex;
+    justify-content: center;
+    box-sizing: border-box;
+    align-items: center;
+    @include square($cell-size);
+    background-color: $cell-color;
+    border: 1px solid black;
+    font-size: 30px;
+    text-align: center;
+    transition: color 0.5s, transform 0.5s;
+
+    @for $index from 1 through $days-in-month {
+      &:nth-child(#{$index})::before {
+        content: "#{$index}";
+      }
+    }
+
+    &:hover {
+      cursor: pointer;
+      background-color: pink;
+      transform: translate(0, -20px);
+      transition: 0.5s;
```

Comment:
Remove the redundant 'transition: 0.5s;' on hover, as it's already set in the base class.